src/commands/mod.rs

```diff
@@ -736,6 +753,403 @@ impl DaemonControl {
+            .filter_map(|(_, amt, deriv)| {
+                if let Some((ind, true)) = &deriv {
+                    Some((*ind, amt))
+                } else {
+                    None
+                }
+            })
+            .max_by(|(ind_1, amt_1), (ind_2, amt_2)| amt_1.cmp(amt_2).then(ind_1.cmp(ind_2)))
+            .map(|(ind, _)| ind);
+        // Use all previous outputs as destinations, except for the output corresponding to the change index we found above.
+        let destinations: HashMap<bitcoin::Address, bitcoin::Amount> = prev_derivs
+            .iter()
+            .filter_map(|(addr, amt, deriv)| match deriv {
+                Some((ind, true)) if *ind == change_index.expect("change_index is some here") => {
+                    None
+                }
+                _ => Some((addr.clone(), *amt)),
+            })
+            .collect();
+
+        // If `!is_cancel`, we take the previous coins as mandatory candidates and add confirmed coins as optional.
+        // Otherwise, we take the previous coins as optional candidates and let coin selection find the
+        // best solution that includes at least one of these. If there are insufficient funds to create the replacement
+        // transaction in this way, then we set candidates in the same way as for the `!is_cancel` case.
+        let mut candidate_coins: Vec<CandidateCoin> = prev_coins
+            .values()
+            .map(|c| CandidateCoin {
+                coin: *c,
+                must_select: !is_cancel,
+            })
+            .collect();
+        let confirmed_cands: Vec<CandidateCoin> = db_conn
+            .coins(&[CoinStatus::Confirmed], &[])
+            .into_values()
+            .filter_map(|c| {
+                // In case the user attempts RBF before the previous coins have been updated in the DB,
+                // they would be returned as confirmed and not spending.
+                if !prev_coins.contains_key(&c.outpoint) {
+                    Some(CandidateCoin {
+                        coin: c,
+                        must_select: false,
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect();
+        if !is_cancel {
+            candidate_coins.extend(&confirmed_cands);
+        }
+        // Try with increasing feerate until fee paid by replacement transaction is high enough.
+        // Replacement fee must be at least:
+        // sum of fees paid by original transactions + incremental feerate * replacement size.
+        // Loop will continue until either we find a suitable replacement or we have insufficient funds.
+        let mut rbf_vsize = 0_u64;
+        loop {
+            // Note that `rbf_vsize` strictly increases on each iteration as otherwise `min_fee`
+            // will ensure that the PSBT meets the required replacement fee and the loop will exit.
+            let min_fee = mempool_entry.fees.descendant.to_sat() + rbf_vsize;
```

Comment:
Why start at 0? This will always fail.

Reply:
I think with 0 we could still get a replacement with a high enough fee due to the higher feerate, and also in case the change output falls below the dust threshold and is added to fees.

Reply:
Fair enough.